plugins/webp-uploads/settings.php

```diff
@@ -12,11 +12,26 @@
 }
 
 /**
- * Registers setting for generating both JPEG and WebP versions for image uploads.
+ * Registers setting for generating JPEG in addition to the selected modern format for image uploads.
  *
  * @since 1.0.0
+ * @since n.e.x.t The setting was made more general to coveroutputting JPEG as a secondary type. The "webp" option naming
+ *        was left unchanged for backward compatibility.
+ * @since n.e.x.t The `perflab_generate_avif_and_webp` was added to enable selecting between AVIF and WebP output.
  */
 function webp_uploads_register_media_settings_field() {
+	if ( wp_image_editor_supports( array( 'mime_type' => 'image/avif' ) ) ) {
+		register_setting(
+			'media',
+			'perflab_generate_avif_and_webp',
+			array(
+				'type'         => 'string',
+				'default'      => 'avif',
```

Comment:
Should this not default to `webp`?

Reply:
Hmm, AVIF is the better/newer format, why not default to it when available?

Reply:
> I guess because WebP is more widely-supported? 

It only defaults to AVIF if you have it available, otherwise the dropdown is not shown and webp  is used as the default output type.

> So if users update this plugin, then they will all of a sudden start getting AVIF images generated instead of WebP ones, which may not be what they expect

Ah, good point, how about I change it so that only new installs get AVIF by default (if available)?

Reply:
In regards to support, I mean not server support but rather client support (and support in popular web apps).

Reply:
Browser support for AVIF is actually pretty solid: https://caniuse.com/avif. I don't think we should be too concerned with the "support in popular web apps" issue for this plugin.

Reply:
resolving for now